File: content/problems/2086-minimum-number-of-food-buckets-to-feed-the-hamsters/solutions/prefer-right-bucket-greedy/Solution.java
class Solution {
    public int minimumBuckets(String street){
        char[] value=street.toCharArray();
        int answer=0;
        for(int i=0;i<value.length;i++)if(value[i]=='H'){
            if(i>0&&value[i-1]=='B')continue;
            if(i+1<value.length&&value[i+1]=='.'){
                value[i+1]='B';
                answer++;
            }else if(i>0&&value[i-1]=='.'){
                value[i-1]='B';
                answer++;
            }else return -1;
        }
        return answer;
    }
}

File: content/problems/2086-minimum-number-of-food-buckets-to-feed-the-hamsters/solutions/prefer-right-bucket-greedy/solution.cpp
class Solution {
    public: int minimumBuckets(string street){
        int answer=0;
        for(int i=0;i<(int)street.size();++i)if(street[i]=='H'){
            if(i&&street[i-1]=='B')continue;
            if(i+1<(int)street.size()&&street[i+1]=='.'){
                street[i+1]='B';
                ++answer;
            }else if(i&&street[i-1]=='.'){
                street[i-1]='B';
                ++answer;
            }else return -1;
        }
        return answer;
    }
};
